src/other/broadcasting.jl

```diff
@@ -144,15 +156,22 @@ if isdefined(Base, :dotgetproperty)
 end
 
 function Base.copyto!(lazydf::LazyNewColDataFrame, bc::Base.Broadcast.Broadcasted{T}) where T
+    df = lazydf.df
+    @assert columnindex(df, lazydf.col) == 0
     if bc isa Base.Broadcast.Broadcasted{<:Base.Broadcast.AbstractArrayStyle{0}}
         bc_tmp = Base.Broadcast.Broadcasted{T}(bc.f, bc.args, ())
         v = Base.Broadcast.materialize(bc_tmp)
-        col = similar(Vector{typeof(v)}, nrow(lazydf.df))
+        col = similar(Vector{typeof(v)}, nrow(df))
         copyto!(col, bc)
     else
         col = Base.Broadcast.materialize(bc)
     end
-    lazydf.df[!, lazydf.col] = col
+    if df isa DataFrame
+        return df[!, lazydf.col] = col
+    else
+        @assert df isa SubDataFrame && getfield(df, :colindex) isa Index
+        return df[:, lazydf.col] = col
```

Comment:
Why use `:` here?

Reply:
because `df` here is a `SubDataFrame`, and we want to create a new column (and in the current version of the PR `!` was not supported - I will change it when we add newely agreed rules).